### PR TITLE
Robot shooter

### PR DESCRIPTION
**Pull request by avidraccoon**

### CODE REVIEW

File: src/main/java/frc/robot/RobotContainer.java
Comment (jkleiber) [inline src/main/java/frc/robot/RobotContainer.java]: We should be implementing the IO classes in separate files - one for hardware and one for sim. They both should use the interface you've created.

File: src/main/java/frc/robot/Robot.java
Comment (jkleiber) [inline src/main/java/frc/robot/Robot.java]: Is this just test code?

File: src/main/java/frc/robot/Robot.java
Comment (jkleiber) [inline src/main/java/frc/robot/Robot.java]: Do you need pose for the shooter?

File: src/main/java/frc/robot/Robot.java
Comment (jkleiber) [inline src/main/java/frc/robot/Robot.java]: can we remove this test code before we merge this?

File: src/main/java/frc/robot/subsystems/shooter_intake/ShooterIntakeSubsystem.java
Comment (jkleiber) [inline src/main/java/frc/robot/subsystems/shooter_intake/ShooterIntakeSubsystem.java:32]: After looking at this more, I'm not sure using the enum functions really is buying us much. It's nice that we get some inheritance here, but it's also sort of confusing to read with `changeStates` and `changeState` having two different outcomes.

One thing that I think is unfortunate is that we have to have a static instance of the subsystem inside the enum. This is not intuitive and I don't think we want to design things this way.

I think we should go back to the system we did last year for now - an enum that defines state and a switch statement that dispatches specific functions which each run the logic for their respective state.

**Review by jkleiber — COMMENTED** [review]
First pass review of the shooter

File: src/main/java/frc/robot/Robot.java
Comment (avidraccoon) [reply src/main/java/frc/robot/Robot.java]: No sorry

File: src/main/java/frc/robot/Robot.java
Comment (avidraccoon) [reply src/main/java/frc/robot/Robot.java]: Yes

File: src/main/java/frc/robot/Robot.java
Comment (avidraccoon) [reply src/main/java/frc/robot/Robot.java]: Yes let me remove it quickly.

File: src/main/java/frc/robot/RobotContainer.java
Comment (avidraccoon) [reply src/main/java/frc/robot/RobotContainer.java]: Forgot to push new commit

File: src/main/java/frc/robot/subsystems/shooter_intake/ShooterIntakeSubsystem.java
Comment (avidraccoon) [reply src/main/java/frc/robot/subsystems/shooter_intake/ShooterIntakeSubsystem.java:32]: I believe I can fix both of those issue very quickly if there is not other issues is it fine to keep this implementation with a bit of cleaning up.

File: src/main/java/frc/robot/RobotContainer.java
Comment (avidraccoon) [reply src/main/java/frc/robot/RobotContainer.java]: I just forgot to remove those lines of code I believe.

**Comment by avidraccoon** [comment]
I'm having problems with solving the conflicting files can you help me?

**Comment by jkleiber** [comment]
@avidraccoon what I do when I see merge conflicts is try to preserve the code that is in conflict with my code while still making sure my feature works. In this case it probably makes sense to "Accept Both" changes when doing the rebase and then refactor the code as needed to make things work.

The rebase process can be found here: https://team401.github.io/Docs/Programming/Software-Workflow.html#keep-branch-updated

If there's a specific problem I can help with just let me know!

File: src/main/java/frc/robot/subsystems/shooter_intake/ShooterIntakeSubsystem.java
Comment (jkleiber) [inline src/main/java/frc/robot/subsystems/shooter_intake/ShooterIntakeSubsystem.java]: Let's move these to constants

We can store constants in src/main/java/frc/robot/constants and eventually JSONify everything

File: src/main/java/frc/robot/subsystems/shooter_intake/ShooterIntakeIOSim.java
Comment (jkleiber) [inline src/main/java/frc/robot/subsystems/shooter_intake/ShooterIntakeIOSim.java]: If these are going to be the control gains, we should move them to constants as well

**Review by jkleiber — COMMENTED** [review]
Second pass, this looks pretty clean now. We just need to get the merge conflicts resolved and then it'll be ready to merge.

After we merge this, we need to get the hardware IO made so we can test on the robot. Can you create a ticket to do this if you haven't already?

**Comment by avidraccoon** [comment]
I moved the constants and prepared for merge but now it can't build because it can find classes for import, how do I fix this?

Edit: It can build but simulation will not run.

Edit 2: Fixed it just needed to clear the build cache

**Review by jkleiber — APPROVED** [review]
There's some refactoring to do here, but let's do that in a separate issue. Approved - please merge when you are able

**Pull request merged**
